src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -205,49 +205,50 @@ string CARMACloudPlugin::updateTags(string str,string tagout, string tagin)
 
 void CARMACloudPlugin::CARMAResponseHandler(QHttpEngine::Socket *socket)
 {
-	QString st; 
+	QByteArray st; 
 	while(socket->bytesAvailable()>0)
 	{	
 		auto readBytes = socket->readAll();
-		if (socket->headers().keys().contains(CONTENT_ENCODING_KEY) && std::string(socket->headers().constFind(CONTENT_ENCODING_KEY).value().data()) == CONTENT_ENCODING_VALUE)
-        {
-			//readBytes is compressed in gzip format
-            st.append(UncompressBytes(readBytes));			
-        }else{
-			st.append(readBytes);
-		}
+		st.append(readBytes);
 	}
-	QByteArray array = st.toLocal8Bit();
-
-	char* _cloudUpdate = array.data(); // would be the cloud update packet, needs parsing
-	
-	
-	string tcm = _cloudUpdate;
 
-	PLOG(logINFO) << "Received TCM from cloud" << tcm << std::endl;
-	if(tcm.length() == 0)
+	if(st.size() == 0)
 	{
-		PLOG(logERROR) << "Received TCM length is zero, and skipped." << std::endl;
+		PLOG(logERROR) << "Received TCM is empty, and skipped." << std::endl;
 		return;
 	}
+	PLOG(logINFO) << "Received TCM bytes size: " << st.size()<< std::endl;
+
+	string tcm = "";
+	bool isCompressed = socket->headers().keys().contains(CONTENT_ENCODING_KEY) && std::string(socket->headers().constFind(CONTENT_ENCODING_KEY).value().data()) == CONTENT_ENCODING_VALUE;
+	if (isCompressed)
+	{
+		tcm = UncompressBytes(st).data();
+	}else{
+		tcm = st.data();		
+	}
 
 	//Transform carma-cloud TCM XML to J2735 compatible TCM XML by updating tags
 	tcm=updateTags(tcm,"<TrafficControlMessage>","<TestMessage05><body>");
 	tcm=updateTags(tcm,"</TrafficControlMessage>","</body></TestMessage05>");
 	tcm=updateTags(tcm,"TrafficControlParams","params");
 	tcm=updateTags(tcm,"TrafficControlGeometry","geometry");
 	tcm=updateTags(tcm,"TrafficControlPackage","package");
-	
-	//List of tcm in string format
-	std::list<std::string> tcm_sl = FilterTCMs(tcm);	
+
+	std::list<std::string> tcm_sl = {};
+	if (isCompressed)
+	{
+		tcm_sl = FilterTCMs(tcm);
```

Comment:
Is there a reason we only call FilterTCMs if the message was originally compressed? If so, it's probably worth at least a comment.

Reply:
The compressed file has list of TCMs wrapped inside the <TrafficControlMessageList> tags. All these are in a big XML payload. This filter tcms is to remove the list tags and only return a list a TCMs objects. There are comments in the header file: https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/625800ebaa77466a5e2eb7a584b57e7f9bce263e/src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.h#L169

Reply:
So if not compressed it is guranteed to be a list of a single TCM but if it is compressed it is multiple TCMs in a single XML that have to be parsed out individually? Is that correct?

Reply:
So then what is this configuration parameter for https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/0e1ee6502a6591769a59e4c08d88262d89aa3d52/src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp#L94